### PR TITLE
Head Decap Shader and Functions Update

### LINKED ISSUE #5465: Fix first-person head decapitation for all avatars
Some avatar models are missing head decapitation when in first-person mode.

### PR DESCRIPTION
**Pull request by mohsenheydari**
## Summary

 Fixes #5465 
 Adds multi-bone support to head decap shader.

## Checklist
- [ ] Pre-push checks pass `npm run check`
  - [ ] Linter passing via `npm run lint`
  - [ ] Unit & Integration tests passing via `npm run test:packages`
  - [ ] Docker build process passing via `npm run build-client`
- [ ] If this PR is still a WIP, convert to a draft 
- [ ] When this PR is ready, mark it as "Ready for review"
- [ ] Changes have been manually QA'd
- [ ] Changes reviewed by at least 2 approved reviewers

## References

_References to pertaining issue(s)_

## QA Steps

1. `git checkout pr_branch_name`
2. `npm install`
3. `npm run dev-reinit`
4. `npm run dev`

List any additional steps required to QA the changes of this PR, as well as any supplemental images or videos.

## Reviewers

_Reviewers for this PR_
@speigg

### CODE REVIEW

**Comment by speigg** [comment]
When the camera gets close, things still get a bit scary looking; would be better if inside faces weren't shown. image

**Comment by zulqarnainhanif** [comment]
@mohsenheydari now this is happening if stand close to a wall, tested this in apartment location.

Screenshot 2022-04-13 at 11 02 58 PM

File: packages/engine/src/avatar/functions/avatarFunctions.ts [hunk 1]
```diff
@@ -276,23 +277,84 @@ export function getDefaultSkeleton(): SkinnedMesh {
   return skinnedMesh
 }
 
+/**
+ * Adds required parameters to mesh's material
+ * to enable avatar's head decapitation (opacity fade)
+ * @param model
+ * @param material
+ * @returns
```
Comment (dmitrizagidulin) [inline packages/engine/src/avatar/functions/avatarFunctions.ts:285]:
```suggestion
```
(no need for a blank `@returns`)

**Comment by mohsenheydari** [comment]
@zulqarnainhanif I'm aware of the issue but could not come up with any fixes for it so far.

**Comment by speigg** [comment]
> @zulqarnainhanif I'm aware of the issue but could not come up with any 'shader' fixes so far. The simplest fix I can think of is to reduce the fade distance as much as possible.

That might be a good option, I'll play with it in a bit

**Pull request merged**